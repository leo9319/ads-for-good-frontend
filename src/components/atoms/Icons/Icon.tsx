import React from 'react';
import { classNames } from '@utils/common/classNames';
import { Link, LinkProps } from '@radix-styles/atoms/Link';

export interface IconProps {
  name: string;
  size?: string;
  iconWidth?: string;
  iconHeight?: string;
  color?: string;
  className?: string;
  backgroundColor?: string;
  backgroundSize?: string;
  backgroundPadding?: string;
  rounded?: boolean;
  style?: React.CSSProperties;
  link?: LinkProps;
  onClick?: () => void;
}
export const Icon = ({
  name,
  size,
  iconWidth,
  iconHeight,
  color,
  className,
  backgroundColor,
  backgroundSize,
  backgroundPadding,
  rounded,
  style,
  link,
  onClick,
}: IconProps) => {
  const linkProps =
    typeof link === 'object' && !(link instanceof Array)
      ? link
      : { ignoreHref: true };
  const sizeStyle = size && `${size}px`;
  return (
    <Link {...linkProps}>
      <span
        className={className}
        style={{
          background: backgroundColor,
          borderRadius: rounded ? '50%' : '0',
          width: backgroundSize,
          height: backgroundSize,
          padding: backgroundPadding,
          display: 'grid',
          alignItems: 'center',
          justifyContent: 'center',
          ...style,
        }}
        onClick={onClick}
      >
        <span
          className={classNames('icon', name)}
          style={{
            width: iconWidth ? `${iconWidth}px` : sizeStyle,
            height: iconHeight ? `${iconHeight}px` : sizeStyle,
            color: color,
          }}
        ></span>
      </span>
    </Link>
  );
};

export interface IconDisplayProps {
  icons: string[];
}

export default Icon;
