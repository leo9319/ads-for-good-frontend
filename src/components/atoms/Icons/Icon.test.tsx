import React from 'react';
import Icon from './Icon';
import { render } from '@testing-library/react';

describe('Icon', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Icon name="icon-eye" />);
    expect(baseElement).toMatchSnapshot();
  });
});
