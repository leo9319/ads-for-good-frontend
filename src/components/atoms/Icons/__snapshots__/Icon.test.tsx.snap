// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render successfully 1`] = `
<body>
  <div>
    <span
      style="border-radius: 0; display: grid; align-items: center; justify-content: center;"
    >
      <span
        class="icon icon-eye"
      />
    </span>
  </div>
</body>
`;
